Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -1,95 +1,183 @@
+            {
+                for (int x = i; x > 0; x--)
+                {
+                    if (array[x] < array[x - 1])
+                    {
+                        int temp = array[x];
+                        array[x] = array[x - 1];
+                        array[x - 1] = temp;
+                    } 
+                    else {
+                        break;
+                    }
+                }
+            }
         }
 
-        /// <summary>
-        /// Reverse the array elements, first being last and so on.
-        /// If array empty or null- don't do anything.
-        /// </summary>
-        /// <param name="array">Input array in a random order.</param>
         public static void Reverse(int[] array)
         {
-            // ToDo: implement.
+            if (IsNullOrEmpty(array))
+            {
+                return;
+            }
+
+            for (int i = 0; i < array.Length/2; i++)
+            {
+                int temp = array[i];
+                array[i] = array[array.Length - 1 - i];
+                array[array.Length -1 - i] = temp;
+            }
+
         }
 
-        /// <summary>
-        /// Remove last element in array.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <returns>A new array with the last element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveLast(int[] array)
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return array;
+            }
+
+            int[] newArray = new int[array.Length - 1];
+            
+            for (int i = 0; i < array.Length - 1; i++)
+            {
+                newArray[i] = array[i];
+            }
+            
+            return newArray;
         }
 
-        /// <summary>
-        /// Remove first element in array.
-        /// </summary>
-        /// <returns>A new array with the first element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveFirst(int[] array)
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return array;
+            }
+
+            int[] newArray = new int[array.Length - 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i + 1];
+            }
+
+            return newArray;
         }
 
-        /// <summary>
-        /// Removes array element at given index.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="index">Index at which array element should be removed.</param>
-        /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
```

Comment:
This method should be returning the original array unmodified if the index is out of range. Consider handling that at the beginning of the method.